src/c_wrappers.jl

```diff
@@ -148,23 +217,62 @@ function add_nomad_array_of_double_param!(prob::C_NomadProblem, keyword::String,
 end
 
 function solve_nomad_problem(prob::C_NomadProblem, x0s::Vector{Float64}, nb_starting_points::Int)
-    # will be used to check if the algorithm finds a feasible or infeasible solution
-    exists_feas_sol = Ref{Cint}(0)
-    exists_inf_sol = Ref{Cint}(0)
-
-    # final values
-    x_feas_sol = zeros(Float64, prob.nb_inputs)
-    x_inf_sol = zeros(Float64, prob.nb_inputs)
-    outputs_feas_sol = zeros(Float64, prob.nb_outputs)
-    outputs_inf_sol = zeros(Float64, prob.nb_outputs)
+    c_result = create_c_nomad_result()
+
+    # Solve problem
     statusflag = ccall((:solveNomadProblem, libnomadCInterface), Cint,
-                       (Ptr{Cvoid}, Cint, Ptr{Float64}, # internal data, number of starting points, starting points,
-                        Ptr{Cint}, Ptr{Float64}, Ptr{Float64}, # feasible solution flag, x_feas, outputs feas
-                        Ptr{Cint}, Ptr{Float64}, Ptr{Float64}, # infeasible solution flag, x_inf, output inf
+                       (Ptr{Cvoid}, Ptr{Cvoid}, # C_NomadResult internal data, C_NomadProblem internal data
+                        Cint, Ptr{Float64}, # number of starting points, starting points,
                         Any),
-                       prob.ref, nb_starting_points, x0s,
-                       exists_feas_sol, x_feas_sol, outputs_feas_sol,
-                       exists_inf_sol, x_inf_sol, outputs_inf_sol, prob)
-    return (Bool(exists_feas_sol[]), x_feas_sol, outputs_feas_sol,
-            Bool(exists_inf_sol[]), x_inf_sol, outputs_inf_sol)
+                       c_result.ref, prob.ref, nb_starting_points, x0s, prob)
+
+    # Collect information
+    result_infos = Dict()
+    result_infos[:status] = statusflag
+    result_infos[:x_sol] = nothing
+    result_infos[:bbo_sol] = nothing
```

Comment:
@amontoison If you have a better name for these attributes, feel free to to tell me. I will change them

Reply:
`statistics` instead of `result_infos`?